### PR TITLE
improved and internationalized ingredients and allergens parsing (French, Spanish, English)

### PR DESCRIPTION
**Pull request by stephanegigandet**
bug #2023

### CODE REVIEW

**Comment by stephanegigandet** [comment]
I started to better internationalize the ingredient parsing and allergens parsing, with improvements for French and Spanish. The code is better structured to add more languages (especially languages that have a similar structure like Italian or Portuguese) for parsing things like "May contain traces of X, Y and Z" etc.

We recognize more allergens, but we also remove the traces from the ingredients_tags, which should help with the rate of ingredients recognition.

We can add support for more cases and languages, but we have to do it slowly and carefully, with new tests for every new pattern, to make sure we don't break things later. It's not just adding translations for helper words, stopwords etc. as that can easily break things.

File: taxonomies/ingredients.txt [hunk 1]
```diff
@@ -23781,6 +23849,51 @@ nl:frambozenaroma
...
+fr:arôme de laurier
+
+<en:flavouring
+en:crustaceans flavouring
+fr:arôme de crustacés
+
+<en:flavouring
+fr:arôme de gentiane
+
+/oran
```
Comment (teolemon) [inline taxonomies/ingredients.txt]: typo @st

Comment (teolemon) [reply taxonomies/ingredients.txt]: @stephanegigandet

Comment (stephanegigandet) [reply taxonomies/ingredients.txt]: Fixed, thanks.

**Comment by teolemon** [comment]
nl: `Kan sporen bevatten van noten, gluten, sesam. `

**Comment by teolemon** [comment]
pt: Pode conter vestígios de frutos de casca rija.

**Comment by teolemon** [comment]
https://www.linguee.com/english-portuguese/search?query=may+contain+traces+of+nuts

**Comment by teolemon** [comment]
Spanish - Feb 7th 2020
Type | Unique tags | Occurrences
-- | -- | --
known | 1939 (4.44%) | 165842 (72.05%)
unknown | 41761 (95.56%) | 64335 (27.95%)
all | 43701 (100.00%) | 230177 (100.00%)

**Pull request merged**